js/tools/flowers.js

```diff
@@ -0,0 +1,39 @@
+/* global C2D */
+window.tools.flowers = {
+  name: 'flowers',
+  icon: '/images/new-flower-icon.png',
+  state: {
+    selected: false,
+    mousePressed: false
+  },
+  events: {
+    mousedown: function () {
+      const state = window.tools.flowers.state
+      state.mousePressed = true
+    },
+    mouseup: function () {
+      const state = window.tools.flowers.state
+      state.mousePressed = false
+    },
+    mousemove: function (e) {
+      const state = window.tools.flowers.state
+      if (state.selected && state.mousePressed) {
+        const mouse = C2D.eventToMouse(e)
+        const scale = 3
+        //  center of flower
+        C2D.fill = 'rgba(235, 219, 52)'
+        C2D.ellipse(mouse.x, mouse.y, 10 * scale + 3)
+        //  flower petals
+        C2D.fill = 'rgba(208, 85, 230)'
```

Comment:
do u remember what we said in class coupe weeks back about changing the fill or stroke color? u need to make sure to change the color back to whatever the user had set beforehand. so u'll need to add a new property to ur state, something like:
```js
state: {
    selected: false,
    mousePressed: false,
    prevColor: null // to keep track of the "previous color"
  },
```
then when the user clicks the mouse down u'll store the canvas's current fill color. when the user moves their mouse u'll change the color to what u need, but then when the user releases the mouse u can use that `prevColor` value to *restore* the canvas back to that color
```js
mousedown: function () {
  const state = window.tools.flowers.state
  state.mousePressed = true
  state.prevColor = C2D.fill // store initial color
},
mouseup: function () {
  const state = window.tools.flowers.state
  state.mousePressed = false
  C2D.fill = state.prevColor // restore back to initial color
},
```